pq/error.go

```diff
@@ -17,16 +17,116 @@
 
 package pq
 
-import "errors"
-
-var (
-	errNODelegate          = errors.New("delegate must not be nil")
-	errInvalidPagesize     = errors.New("invalid page size")
-	errClosed              = errors.New("queue closed")
-	errNoQueueRoot         = errors.New("no queue root")
-	errIncompleteQueueRoot = errors.New("incomplete queue root")
-	errInvalidVersion      = errors.New("invalid queue version")
-	errACKEmptyQueue       = errors.New("ack on empty queue")
-	errACKTooManyEvents    = errors.New("too many events have been acked")
-	errSeekPageFailed      = errors.New("failed to seek to next page")
+import (
+	"github.com/elastic/go-txfile"
+
+	"github.com/elastic/go-txfile/internal/strbld"
+	"github.com/elastic/go-txfile/txerr"
+)
+
+// ErrKind provides the pq related error kinds
+type ErrKind int
+
+type reason interface {
+	txerr.Error
+}
+
+type Error struct {
```

Comment:
Does each package need it's own definition of `Error`? They are not 100% identical but look very similar.